jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/MultiInstanceCallActivityRuntimeManagerTest.java

```diff
@@ -0,0 +1,157 @@
+import org.jbpm.runtime.manager.util.TestUtil;
+import org.jbpm.services.task.identity.JBossUserGroupCallbackImpl;
+import org.jbpm.test.util.AbstractBaseTest;
+import org.jbpm.test.util.CountDownProcessEventListener;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.event.process.ProcessEventListener;
+import org.kie.api.io.ResourceType;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.manager.RuntimeEnvironment;
+import org.kie.api.runtime.manager.RuntimeEnvironmentBuilder;
+import org.kie.api.runtime.manager.RuntimeManager;
+import org.kie.api.runtime.manager.RuntimeManagerFactory;
+import org.kie.api.runtime.manager.audit.ProcessInstanceLog;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.internal.io.ResourceFactory;
+import org.kie.internal.runtime.manager.context.ProcessInstanceIdContext;
+import org.kie.internal.task.api.UserGroupCallback;
+
+import bitronix.tm.resource.jdbc.PoolingDataSource;
+
+public class MultiInstanceCallActivityRuntimeManagerTest extends AbstractBaseTest {
+    
+    private PoolingDataSource pds;
+    private UserGroupCallback userGroupCallback;
+    private EntityManagerFactory emf;
+    private RuntimeManager manager;
+    
+    private CountDownProcessEventListener countDownListener;
+    
+    private int numberOfChildProcesses = 15;
+    
+    @Before
+    public void setup() {
+        TestUtil.cleanupSingletonSessionId();
+        pds = TestUtil.setupPoolingDataSource();
+        emf = EntityManagerFactoryManager.get().getOrCreate("org.jbpm.persistence.jpa");
+        Properties properties= new Properties();
+        properties.setProperty("mary", "HR");
+        properties.setProperty("john", "HR");
+        userGroupCallback = new JBossUserGroupCallbackImpl(properties);
+    }
+    
+    @After
+    public void teardown() {
+        if (manager != null) {
+            manager.close();
+        }
+        EntityManagerFactoryManager.get().clear();
+        pds.close();
+    }
+    
+    @Test(timeout=10000)
+    public void testSingletonMultiInstanceCallactivityCompleteAtTheSameTime() {
+        RuntimeEnvironment environment = createEnvironment();        
+        manager = RuntimeManagerFactory.Factory.get().newSingletonRuntimeManager(environment, "first");        
+        assertNotNull(manager);
+        
+        testMultiInstanceCallactivityCompleteAtTheSameTime();
+    }
+    
+    @Test(timeout=10000)
+    public void testPerProcessInstanceMultiInstanceCallactivityCompleteAtTheSameTime() {
+        RuntimeEnvironment environment = createEnvironment();        
+        manager = RuntimeManagerFactory.Factory.get().newPerProcessInstanceRuntimeManager(environment, "first");        
+        assertNotNull(manager);
+
+        testMultiInstanceCallactivityCompleteAtTheSameTime();
+    }
+
+
+    public void testMultiInstanceCallactivityCompleteAtTheSameTime() {
```

Comment:
I would make this `private` as now it slightly looks like this could a be a test method with a missing `@Test` annotation.